src/app/ApplicationTemplate.Access/ApiClients/Authentication/AuthenticationApiClientMock.cs

```diff
@@ -45,41 +49,58 @@ public async Task<AuthenticationData> RefreshToken(CancellationToken ct, Authent
 
 		await SimulateDelay(ct);
 
-		return CreateAuthenticationData(unauthorizedToken.AccessTokenPayload);
-	}
-
-	private AuthenticationData CreateAuthenticationData(AuthenticationToken token = null, TimeSpan? timeToLive = null)
-	{
-		var encodedJwt = CreateJsonWebToken(token, timeToLive);
-		var jwt = new JwtData<AuthenticationToken>(encodedJwt, _serializerOptions);
-
-		return new AuthenticationData()
-		{
-			AccessToken = jwt.Token,
-			RefreshToken = Guid.NewGuid().ToString(format: null, CultureInfo.InvariantCulture),
-			Expiration = jwt.Payload.Expiration,
-		};
+		return CreateAuthenticationData(token: unauthorizedToken.AccessToken.Payload);
 	}
 
-	private string CreateJsonWebToken(AuthenticationToken token = null, TimeSpan? timeToLive = null)
+	/// <summary>
+	/// Creates a JSON Web Token.
+	/// </summary>
+	/// <remarks>
+	/// This function has been made public and static for testing purposes.
+	/// </remarks>
+	/// <param name="token">The token to use.</param>
+	/// <param name="email">The email or unique name to store in the token.</param>
+	/// <param name="now">The current date and time to use for the authentication token.</param>
+	/// <param name="serializerOptions">The serializer options to use for the token serialization.</param>
+	/// <returns>The JSON Web token.</returns>
+	public static string CreateJsonWebToken(AuthenticationToken token = null, string email = null, DateTimeOffset? now = null, JsonSerializerOptions serializerOptions = null)
 	{
 		const string header = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9"; // alg=HS256, type=JWT
 		const string signature = "QWqnPP8W6ymexz74P6quP-oG-wxr7vMGqrEL8y_tV6M"; // dummy stuff
 
-		var now = DateTimeOffset.Now;
+		now ??= DateTimeOffset.Now;
```

Comment:
Why not use `_timeProvider` (which is already used below)?

Reply:
This is a static method.